homeassistant/components/tile/device_tracker.py

```diff
@@ -65,52 +83,56 @@ class TileDeviceTracker(CoordinatorEntity, TrackerEntity):
 
     _attr_icon = DEFAULT_ICON
 
-    def __init__(self, entry, coordinator, tile):
+    def __init__(
+        self, entry: ConfigEntry, coordinator: DataUpdateCoordinator, tile: Tile
+    ) -> None:
         """Initialize."""
         super().__init__(coordinator)
 
-        self._attr_extra_state_attributes = {ATTR_ATTRIBUTION: DEFAULT_ATTRIBUTION}
+        self._attr_extra_state_attributes: MutableMapping[str, Any] = {
```

Comment:
Without this re-casting, `mypy` complains:

```
homeassistant/components/recollect_waste/sensor.py:121: error: Item "None" of "Optional[MutableMapping[str, Any]]" has no attribute "update"  [union-attr]
```

Perhaps because the base entity attribute allows for `None`?

https://github.com/home-assistant/core/blob/edf42bab25ae2e3681f3d0e774b7eebdaaa87ce3/homeassistant/helpers/entity.py#L230

Reply:
With #53529 merged, if you rebase this PR, you can remove the type hint.